### DISCUSSION TITLE
Business Favorites

### DISCUSSION THREAD

**Discussion by sarvaiyanidhi** [joemasilotti/railsdevs.com#878] [Ideas]
The idea of allowing subscribed businesses to mark specific profiles as favorites for potential future collaboration seems promising. By limiting the number of favorites to a reasonable amount, such as 5, it helps ensure that businesses are selective in their choices and avoid overcrowding the site with excessive favorites.

Having a favorite list for businesses allows them to keep track of developers they believe are a good fit for their needs. Even if these developers are not currently available for new opportunities, the system could incorporate a feature that sends alerts to businesses via email once the developers become open to new opportunities. This notification system could serve as a valuable tool for businesses to reach out to potential candidates at the right time, increasing the chances of successful collaborations.

Overall, implementing this feature can enhance the user experience for businesses and developers, facilitating more targeted and efficient connections between the two parties.

**Comment by joemasilotti** [comment]
I love the idea of _shortlisting_ candidates in RailsDevs. I do worry a bit about privacy concerns with the "watching" of status, though. I want to think on it a bit more.

Do you have appetite to build an MVP of this out? An initial version could consist of:

1. Businesses can click "watch" on a developer profile
2. `/developers/watched` as the index view for the business
3. Notifications when the developer's status changes (perhaps not an MVP feature, though)
4. Add statuses to each developer (not contacted, contacted, replied, etc.) - this could also pave the way for tracking where they are in the application process.

**Comment by sarvaiyanidhi** [comment]
> I love the idea of shortlisting candidates in RailsDevs. I do worry a bit about privacy concerns with the "watching" of status, though. I want to think on it a bit more.

May be we can provide option in developer's profile with checkbox if they want to allow their profile to get watched by business or not to handle privacy concern and providing option willingly. So, only profiles who has enabled option will have watch button on profile. Also, developer cannot uncheck if any business is watching after initial setting. I think it may add some complexity and checks but can handle feature corretly.

> Add statuses to each developer (not contacted, contacted, replied, etc.) - this could also pave the way for tracking where they are in the application process.

Can you clarify more on this point as already I think conversations capture any communication between developer and business so do we want to add status additionally.

I will work on MVP and create PR soon. Do you want me to create issue or can I just go ahead with creating feature and submitting PR.

**Comment by matteeyah** [comment]
I also think this is a good proposal. When using the product, I use a Google Doc to keep a list of people I want to reach out to, then use that as reference.

This can be progressively enhanced later to track the status of reaching out to a developer, so it's not a part of the MVP.

Also having https://github.com/joemasilotti/railsdevs.com/discussions/888 would increase the value add of this by an order of a magnitude.

**Reply by joemasilotti** [reply]
Awesome to hear there's a need for this! Thanks for chiming in @matteeyah. And welcome to the platform. :)

I'm going to follow up in a private channel to hear more about how _you_ want to use this in the future. Once that's straightened out I'll post the summary here for @sarvaiyanidhi to start work (if they still have appetite, of course).

**Reply by sarvaiyanidhi** [reply]
Thank you, @matteeyah! It's great to hear that you find this feature useful. And @joemasilotti, I'm certainly interested in working on this feature once we have all the details sorted out.

Additionally, I'm thinking that we could consider adding a field for businesses, which would enable a shortlisting capability. This way, we can roll out the feature for selected businesses as needed.
